### ISSUE TITLE
M2M connection is broken

### ISSUE THREAD

**Issue by KaRi94** [graphql-python/graphene-django#172]
Hi,

I have following models:

```
class Place(models.Model):
    categories = models.ManyToManyField(PlaceCategory, related_name='places')
```
```
class PlaceCategory(models.Model):
    name = models.CharField(max_length=255, unique=True)
```
Furthermore, I have following DjangoObjectType models:

```
class PlaceNode(DjangoObjectType):
    class Meta:
        model = Place
        interfaces = (graphene.relay.Node, )
```
```
class PlaceCategoryNode(DjangoObjectType):
    class Meta:
        model = PlaceCategory
        interfaces = (graphene.relay.Node, )
```

Then when I execute this query:
```
{
  placeCategories {
    edges {
      node {
        id
        places {
          edges {
            node {
              id
            }
          }
        }
      }
    }
  }
}
```

everything works great but when I try to execute this one:
```
{
  places {
    edges {
      node {
        id
        categories {
          edges {
            node {
              id
            }
          }
        }
      }
    }
  }
}
```
I get errors: 
> Cannot combine queries on two different base models.

It seems that M2M connection is working improperly and it is unable to fetch related objects from instance where M2M is defined.

Has anyone run into the same problem ?

**Comment by KaRi94** [comment]
Issue seems to be related to: https://github.com/graphql-python/graphene/issues/466

**Comment by qntnrbns** [comment]
Do you have any ideas for a work around? I have just run into this issue.

**Comment by KaRi94** [comment]
You can just define custom resolver and return queryset. It should work.

In my case it would be:
```
class PlaceNode(DjangoObjectType):
    class Meta:
        model = Place
        interfaces = (graphene.relay.Node, )

    categories = DjangoFilterConnectionField(PlaceCategoryNode)

    def resolve_categories(self, *args):
        return self.categories.all()
```

But this is only a workaround. In my case when I have lots of M2M it is annoying that I have to define custom resolvers for all of them.

**Comment by KaRi94** [comment]
It looks that exception is raised by:
```
DjangoConnectionField.merge_querysets
```

Method gets two queryset, `default_queryset` and `queryset`. Although `queryset` contains proper instances from m2m connection,  `default_queryset` contains all objects from base model. In my case `queryset` has PlaceCategory instances while `default_queryset` has Place instances. That's why it raises error.

 Looks to me that `DjangoFilterConnectionField.connection_resolver` receives wrong `default_manager` parameter.

Can anyone confirm this ?

**Comment by spockNinja** [comment]
I believe I have solved this bug in #181. Feel free to try that out and let me know if it works for you.

**Comment by keremgocen** [comment]
@spockNinja is this already in 1.3 release, how can I try it?

**Comment by spockNinja** [comment]
@keremgocen No, this was not merged in time for the 1.3 release. There has not been a new release to pypi since this was merged. From the commit log, it looks like a release is pushed about every month or so, so I don't expect it will be too long.

In the meantime, you can pull down master instead of 1.3.

`pip uninstall graphene-django`
`pip install git+https://github.com/graphql-python/graphene-django.git`

or in requirements.txt
`pip install git+ssh://git@github.com/graphql-python/graphene-django.git#egg=graphene-django`

**Comment by clodal** [comment]
Passing by. #181 is still not available at the time of this posting.

The requirements.txt version spockNinja provided didn't work for me.  The following did instead:

```
// requirements.txt
...
graphene==1.4.1
git+git://github.com/graphql-python/graphene-django.git#egg=graphene-django // <<<
graphql-core==1.1
...
```

Also remember to do `pip uninstall graphene-django` and `pip install -r requirements.txt` before and after making the change respectively. M2M FTW!
